### PR TITLE
(feat) [SD-1461] Mark content not for publication

### PR DESCRIPTION
**Pull request by sivakuna-aap**

### CODE REVIEW

File: client/app/scripts/superdesk-archive/views/media-box-grid.html [hunk 1]
```diff
@@ -39,6 +39,9 @@ <h5 ng-class="{'shifted' : item.highlights}">
             <dt ng-show="item.is_spiked" translate>expires</dt>
             <dd ng-if="item.is_spiked"><time title="{{ item.expiry | date:'medium' }}">{{ item.expiry | reldate }}</time></dd>
         </dl>
+
+        <span ng-if="item.marked_for_not_publication == true" class="not_for_publication">Not for publication</span>
```
Comment (amagdas) [inline client/app/scripts/superdesk-archive/views/media-box-grid.html]: translate missing

File: client/app/scripts/superdesk-archive/views/media-box-list.html [hunk 2]
```diff
@@ -32,6 +32,7 @@
             <div class="takekey" ng-if="item.anpa_take_key">{{ item.anpa_take_key }}</div>
             <div class="category" ng-if="item['anpa-category'].name">{{ item['anpa-category'].name }}</div>
             <div class="provider" sd-meta-ingest data-item="item"></div>
+            <div ng-if="item.marked_for_not_publication == true" class="not_for_publication">Not for publication</div>
```
Comment (amagdas) [inline client/app/scripts/superdesk-archive/views/media-box-list.html]: Translate.

File: client/app/scripts/superdesk-authoring/authoring.js [hunk 3]
```diff
@@ -502,11 +503,13 @@
...
 
                 $scope.publish_enabled = $scope.origItem && $scope.origItem.task && $scope.origItem.task.desk &&
-                    ((!_.contains(['published', 'killed'], $scope.origItem.state) && $scope.privileges.publish === 1) ||
-                     ($scope.origItem.state === 'published' && $scope.privileges.kill === 1));
+                    (($scope.privileges.publish === 1 && !authoring.isPublished($scope.origItem)) ||
+                    ($scope.origItem.state === 'published' && $scope.privileges.kill === 1));
 
                 $scope.save_visible = $scope._editable && !authoring.isPublished($scope.origItem);
 
+                $scope.not_for_publication_visisble = $scope.publish_enabled && $scope.origItem.marked_for_not_publication === false;
```
Comment (amagdas) [inline client/app/scripts/superdesk-authoring/authoring.js]: Small typo, s/visisble/visible

File: client/app/scripts/superdesk-authoring/authoring.js [hunk 4]
```diff
@@ -690,6 +694,11 @@
                     }
                 };
 
+                $scope.markForPublication = function(isArticlePublishabble) {
```
Comment (amagdas) [inline client/app/scripts/superdesk-authoring/authoring.js]: s/isArticlePublishabble/isArticlePublishable

**Comment by amagdas** [comment]
:+1:

File: server/apps/archive/common.py [hunk 5]
```diff
@@ -291,10 +291,16 @@ def is_update_allowed(archive_doc):
         raise SuperdeskApiError.forbiddenError("Item isn't in a valid state to be updated.")
 
 
-def set_pub_status(doc, pub_status_value='usable'):
+def handle_existing_data(doc, pub_status_value='usable', doc_type='archive'):
     """
-    Sets the value of pubstatus prooperty in metadata of doc in either ingest or archive repo
+    Handles existing data. For now the below are handled:
+        1. Sets the value of pubstatus prooperty in metadata of doc in either ingest or archive repo
```
Comment (plamut) [inline server/apps/archive/common.py]: Typo here, s/prooperty/property

**Comment by mdhaman** [comment]
:+1:

**Pull request merged**